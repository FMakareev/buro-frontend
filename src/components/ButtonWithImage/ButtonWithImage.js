import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Flex from '../Flex/Flex';
import ButtonBase from '../ButtonBase/ButtonBase';

export const ButtonWithImage = ({ iconLeft, iconRight, children, ...rest }) => (
  <ButtonBase {...rest}>
    <Flex justifyContent="space-between" alignItems="space-around" width="100%">
      {iconLeft && (
        <Flex justifyContent="center" alignItems="center">
          {iconLeft}
        </Flex>
      )}
      <Flex justifyContent="center" alignItems="center">
        {children}
      </Flex>
      )}
      <Flex justifyContent="center" alignItems="center">
        {children}
      </Flex>
      {iconRight && (
        <Flex justifyContent="center" alignItems="center">
          {iconRight}
        </Flex>
      )}
    </Flex>
  </ButtonBase>
);

ButtonWithImage.propTypes = {
  /** name button */
  name: PropTypes.string,
  /** left icon */
  iconLeft: PropTypes.any,
  /** right icon */
  iconRight: PropTypes.any,
  /** variant button */
  variant: PropTypes.string,
  /** size button */
  size: PropTypes.string,
  /** css value - margin-right */
  mr: PropTypes.number,
  /** css value - margin-left */
  ml: PropTypes.number,
};

export default ButtonWithImage;
