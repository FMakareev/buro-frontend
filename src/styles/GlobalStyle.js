import { createGlobalStyle } from 'styled-components';
import BG from '../assets/image/bg.jpg';

export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
    font-family: 'Montserrat Regular';
    font-size: 16px;
  }
  #app{
    background-color: #fff;
    background-image: url(${BG});
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
  }
  a {
    text-decoration: none;
    color: inherit;
  }
  
  button:focus, input:focus, select:focus {
    outline: none !important;
    box-shadow: 0 0 0 5px #E7EDF8;
  }
  

  @font-face {
    font-family: 'Montserrat Regular';
    src: url('../assets/fonts/Montserrat-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
  font-family: 'Montserrat Medium';
  src: url('../assets/fonts/Montserrat-Medium.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
  }

  @font-face {
  font-family: 'Montserrat Bold';
  src: url('../assets/fonts/Montserrat-Bold.ttf') format('truetype');
  font-weight: bold;
  font-style: normal;
}

`;

export default GlobalStyle;
