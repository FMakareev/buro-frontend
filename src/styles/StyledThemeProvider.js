import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { TextVariant } from './variants/textVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#093971',
  color2: '#EFF9FC',
  color3: '#E7EDF8',
  color4: '#61C3E2',
  color5: '#A2A2A2',
  color6: '#0F60BD',
  color7: '#072C57',
  color8: '#61C3E2',
  color9: '#EB5757',
  color10: '#E7EDF8',
  color11: '',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  14,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['576px', '768px', '992px', '1200px'];

const boxShadow = ['4px 8px 16px rgba(28, 65, 84, 0.08)'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        text: TextVariant,
      },
      fontFamily: {
        regular: 'Montserrat Regular',
        medium: 'Montserrat Medium',
        bold: 'Montserrat Bold',
      },
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
