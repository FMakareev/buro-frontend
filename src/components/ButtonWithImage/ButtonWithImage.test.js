import React from 'react';
import renderer from 'react-test-renderer';
// import 'jest-styled-components'; // этот импорт нужен чтобы в снэпшоты попали css стили которые генерируются при рендере компонента
import ArrowLeft from '../../assets/icons/multicolor/arrowLeft.multicolor.svg';
import ArrowRight from '../../assets/icons/multicolor/arrowRight.multicolor.svg';

import {StyledThemeProvider} from '../../styles/StyledThemeProvider';
import {ButtonWithImage} from './ButtonWithImage';


test('ButtonWithImage: Рендерится без ошибок', () => {
  const output = renderer.create(<StyledThemeProvider>
      <ButtonWithImage children={'Button'} variant={'large'} size={'large'}/>
    </StyledThemeProvider>
  ).toJSON();
  expect(output).toMatchSnapshot();
});

test('ButtonWithImage: Рендерится с иконкой справа', () => {
  const output = renderer.create(<StyledThemeProvider>
      <ButtonWithImage rightIcon={<ArrowRight/>} children={'Button'} variant={'large'} size={'large'}/>
    </StyledThemeProvider>
  );
  expect(output).toMatchSnapshot();
});

test('ButtonWithImage: Рендерится с иконкой слева', () => {
  const output = renderer.create(<StyledThemeProvider>
      <ButtonWithImage leftIcon={<ArrowLeft/>} children={'Button'} variant={'large'} size={'large'}/>
    </StyledThemeProvider>
  );
  expect(output).toMatchSnapshot();
});

