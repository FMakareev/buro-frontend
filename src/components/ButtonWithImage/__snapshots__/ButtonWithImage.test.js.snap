// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithImage: Рендерится без ошибок 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <button
    className="mxzgdr-0 bbXgdI"
    size="large"
    type="button"
  >
    <div
      className="sc-1jxwge6-0 wterhv-0 eZGOHC"
      width="100%"
    >
      <div />
      <div
        className="sc-1jxwge6-0 wterhv-0 irkFkV"
      >
        Button
      </div>
      <div />
    </div>
  </button>
</div>
`;

exports[`ButtonWithImage: Рендерится с иконкой слева 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <button
    className="mxzgdr-0 bbXgdI"
    size="large"
    type="button"
  >
    <div
      className="sc-1jxwge6-0 wterhv-0 eZGOHC"
      width="100%"
    >
      <div />
      <div
        className="sc-1jxwge6-0 wterhv-0 irkFkV"
      >
        Button
      </div>
      <div />
    </div>
  </button>
</div>
`;

exports[`ButtonWithImage: Рендерится с иконкой справа 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <button
    className="mxzgdr-0 bbXgdI"
    size="large"
    type="button"
  >
    <div
      className="sc-1jxwge6-0 wterhv-0 eZGOHC"
      width="100%"
    >
      <div />
      <div
        className="sc-1jxwge6-0 wterhv-0 irkFkV"
      >
        Button
      </div>
      <div />
    </div>
  </button>
</div>
`;
